src/loaders/zarr/pixel-source.ts

```diff
@@ -0,0 +1,115 @@
+import { BoundsCheckError } from 'zarr';
+import { isInterleaved } from '../utils';
+import { getIndexer } from './lib/indexer';
+
+import type { HTTPStore } from './lib/storage';
+import type { ZarrArray } from 'zarr';
+import type { RawArray } from 'zarr/dist/types/rawArray';
+import type { AsyncStore } from 'zarr/dist/types/storage/types';
+
+import type {
+  PixelSource,
+  Labels,
+  RasterSelection,
+  PixelSourceSelection,
+  LayerData,
+  TileSelection
+} from '../../types';
+
+const DTYPE_LOOKUP = {
+  u1: 'Uint8',
+  u2: 'Uint16',
+  u4: 'Uint32',
+  f4: 'Float32'
+} as const;
+
+type ZarrIndexer<S extends string[]> = (
+  sel: { [K in S[number]]: number } | number[]
+) => number[];
+
+interface ZarrTileSelection {
```

Comment:
What is this for? Can we reuse the `TileSelection`?

Reply:
Look at the function declaration below in `getTile` Its a union of `TileSelection<S>` and `ZarrTileSelection`. It faithfully implements PixelSource, but allows a selection to be `number[]` regardless of the labels.

Reply:
Are the labels necessary for Tiff?  Could you give an example here?  Just trying to understand.